src/main/java/com/openlattice/postgres/ResultSetAdapters.java

```diff
@@ -766,30 +766,39 @@ public static AppType appType( ResultSet rs ) throws SQLException {
 
     public static Map<UUID, Set<Object>> implicitEntityValuesById(
             ResultSet rs,
-            Map<UUID, PropertyType> authorizedPropertyTypes ) throws SQLException {
+            Map<UUID, PropertyType> authorizedPropertyTypes,
+            ByteBlobDataManager byteBlobDataManager) throws SQLException {
         final Map<UUID, Set<Object>> data = new HashMap<>();
 
         for ( PropertyType propertyType : authorizedPropertyTypes.values() ) {
             List<?> objects = propertyValue( rs, propertyType );
 
             if ( objects != null ) {
-                data.put( propertyType.getId(), new HashSet<>( objects ) );
+                if ( propertyType.getDatatype() == EdmPrimitiveTypeKind.Binary ) {
+                    data.put( propertyType.getId(), new HashSet<>( byteBlobDataManager.getObjects( objects ) ) );
```

Comment:
This should be returning pre-signed urls not the raw objects themselves.

Reply:
It does. The objects stored that are being retrieved are the strings that are keys to the s3 bucket. The getObjects method will generate a presigned url for each key in that list.